crates/fluvio-smartengine/src/engine.rs

```diff
@@ -109,21 +110,26 @@ impl SmartModuleChainInstance {
     pub fn process(
         &mut self,
         input: SmartModuleInput,
+        number_of_records: u64,
         metric: &SmartModuleChainMetrics,
     ) -> Result<SmartModuleOutput> {
         let raw_len = input.raw_bytes().len();
         debug!(raw_len, "sm raw input");
         metric.add_bytes_in(raw_len as u64);
 
         let base_offset = input.base_offset();
-
+        let mut smartmodule_usage_in = number_of_records;
+        let mut smartmodule_usage_out = 0;
         if let Some((last, instances)) = self.instances.split_last_mut() {
+            metric.add_records_in(number_of_records);
             let mut next_input = input;
 
             for instance in instances {
                 // pass raw inputs to transform instance
                 // each raw input may result in multiple records
                 let output = instance.process(next_input, &mut self.store)?;
+                smartmodule_usage_out += output.successes.len() as u64;
+                smartmodule_usage_in += output.successes.len() as u64;
```

Comment:
why increase this? this is output

Reply:
because the output of previous is the input to next

Reply:
notice this is line is not called for the last smartmodule.

Reply:
what's definition of usage then?

Reply:
```
 usage = max(total_records_in, total_records_out) where
 total_records_in = Number of records passed to each smartmodule in the chain.
 total_records_out =Number of records returned from each smartmodule in the chain.
```

Reply:
Suppose we have the following records count of each stage 10 => 5 => 3 => 2,  total amount of work should be just the sum of all inputs = 10+5+3 = 18

Reply:
intent is to capture case where more output than input; 
For example 1 => 10.

Reply:
In that case 10 => 5 => 3 => 10,  so total record processed = 10 + 5 + 10.  So the correct definition is
total usage  = sum ( usage in each stage).   usage in each stage = max(input,output)